### ISSUE TITLE
Allow user-defined pull request number

### ISSUE THREAD

**Issue by xli12** [davelosert/vitest-coverage-report-action#365]
Hi,

We are using this Github Action to report the test coverage in our pipeline. However, the actions are not always triggered by `pull_request`, but `push` instead. As a result, in the context there is no pull request number:

```JavaScript
let pullRequestNumber = github.context.payload.pull_request?.number;
```

Therefore a user-defined pull request number as an input would be helpful (suppose the `prNumber` is given by the user):

```JavaScript
let pullRequestNumber = github.context.payload.pull_request?.number || prNumber;
```

There are other test coverage GitHub Actions which allow this, for example https://github.com/ArtiomTr/jest-coverage-report-action/tree/main.

I could also submit a pull request if needed.

**Comment by davelosert** [comment]
Hey @xli12,

thank you for opening up the issue! 

Definitely something I would like to support, and I would welcome a PR for this very much! 👍

**Comment by xli12** [comment]
Hi @davelosert ,

Thank you for your quick reply! A pull request #369 has been submitted. Please let me know if you have further questions.

**Comment by davelosert** [comment]
@xli12: I just merged the PR and this feature got released in version `2.4.0`. 
Feel free to chime in here and let me know if it works as expected, then we can close the issue.

Thank you again for contributing! ❤

**Comment by xli12** [comment]
@davelosert Thank you for all the assistance with this feature. Just tested using the latest version `v2.4.0` in our repository, it works really well 👍

<img width="1043" alt="Screenshot 2024-06-12 at 10 09 38" src="https://github.com/davelosert/vitest-coverage-report-action/assets/1527106/97a8f4ca-e83c-458e-b6f9-34b5e7ab6d72">

<img width="928" alt="Screenshot 2024-06-12 at 10 10 42" src="https://github.com/davelosert/vitest-coverage-report-action/assets/1527106/2c3462bf-24da-4a80-8b48-9399fbdafcd2">

**Comment by davelosert** [comment]
@xli12 : Thank you so much for the feedback - glad to see it works and already provides value to you! ❤ 

I will close this issue then!

**Comment by davelosert** [comment]
@xli12: Coming back here - I've just [released v2.6.0 of this action](https://github.com/davelosert/vitest-coverage-report-action/releases/tag/v2.6.0) with a feature with this action that allows you to set the `pr-number: "auto"`, which will automatically try to find a pull-request associated with the latest head-sha of the commit on a `push` event. Maybe this is helpful for you, too? 

I guess you have some intermediate action right now that would do the same that you potentially could remove then. Let me know if it's useful and works for you! :)
